src/config/cmake/HYPRE_CMakeUtilities.cmake

```diff
@@ -592,3 +592,65 @@ function(print_option_status)
 
   message(STATUS "")
 endfunction()
+
+# Macro for setting up mixed precision compilation (must be defined before subdirectories)
+macro(setup_mixed_precision_compilation module_name)
+  set(options "")
+  set(oneValueArgs "")
+  set(multiValueArgs SRCS)
+  cmake_parse_arguments(REGULAR "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+  if(NOT REGULAR_SRCS)
+    message(FATAL_ERROR "SRCS argument is required for setup_mixed_precision_compilation")
+  endif()
+
+  # Create object libraries for each precision
+  add_library(${module_name}_flt  OBJECT ${REGULAR_SRCS})
+  add_library(${module_name}_dbl  OBJECT ${REGULAR_SRCS})
+  add_library(${module_name}_ldbl OBJECT ${REGULAR_SRCS})
+  add_library(${module_name}_def  OBJECT ${REGULAR_SRCS})
```

Comment:
This '_def' compile no longer happens in the autotools build, but I forgot to remove it from Makefile.config (there are also a few remnants in 'struct_mv/Makefile' in the 'clean' target.  If you are just mimicking autotools here, then you can remove both.